### DISCUSSION TITLE
Preserve the data inside the New-HTML and use it as Email Body

### DISCUSSION THREAD

**Discussion by swordfish291** [EvotecIT/PSWriteHTML#259] [Q&A] [answered]
Hello

I was trying to get my way around to generate an HTML report and export it as an HTML file and use the data while creating the HTML file to prepare the summary that I want to send as an email body. Is there a way around that? ..

Thank you

**Comment by PrzemyslawKlys** [comment, marked as answer]
Just create 2 different HTML files. There's also -AttachSelf for Emails which attaches "better" version to email with JS enabled. But otherwise you have to create "specific" summary HTML for email, and then create proper report with more information and just attach it as a file.

**Reply by tomerikko** [reply]
i imagine that if he attach it as a file, it will not be in the email Body.
or you think that ps send-mailmessage will take the HTML as body?  i guess not due to all the JS code in there right?

**Reply by PrzemyslawKlys** [reply]
When you create EmailBody { ... } it's HTML without JS, but if you do attachself it uses same code, but with JS included, yet in the email it's still showing the content (just without JS). Then you can attach another version use Attachment which has everything in it.
